chore: Add article about limiting the number of running workspaces ac… (#2788)
codenvy/che-docs: modules/administration-guide/pages/concealing-editors-definitions.adoc

:_content-type: PROCEDURE
:description: Concealing editors definitions
:keywords: administration guide, concealing, dashboard, editors
:navtitle: Concealing editors definitions

[id="concealing-editors-definitions"]
= Concealing editors definitions

> Learn how to conceal {prod-short} editor definitions. It can be useful when you want to hide some of the editors from the Dashboard UI, e.g. IntelliJ and have only Code OSS.

Reviewer: ```suggestion
Learn how to conceal {prod-short} editor definitions. This is useful when you want to hide selected editors from the Dashboard UI, e.g. hide the IntelliJ IDEA Ultimate and have only Visual Studio Code - Open Source visible.
```

.Prerequisites

* An active `{orch-cli}` session with administrative permissions to the {orch-name} cluster. See {orch-cli-link}.

* `jq`. See link:https://stedolan.github.io/jq/download/[Downloading `jq`].

.Procedure

. Find out the namespace where the {prod-short} Operator is deployed:
+
[source,subs="+attributes"]
----
OPERATOR_NAMESPACE=$({orch-cli} get pods -l app.kubernetes.io/component={prod-operator} -o jsonpath={".items[0].metadata.namespace"} --all-namespaces)
----

. Find out the available editors definitions files:
+
[source,subs="+attributes"]
----
{orch-cli} exec -n $OPERATOR_NAMESPACE deploy/{prod-operator} -- ls /tmp/editors-definitions
----

> The output should look similar to the following:

Reviewer: ```suggestion
The output should look similar to the following example:
```

+
[source]
----
che-code-insiders.yaml
che-code-latest.yaml
che-idea-latest.yaml
che-idea-next.yaml
----

. Choose an editor definition to conceal.
For example, to conceal the `che-idea-next.yaml` editor definition, set the editor definition file name:
+
[source,subs="+attributes"]
----
CHE_EDITOR_CONCEAL_FILE_NAME=che-idea-next.yaml
----

. Define the ConfigMap name for the concealed editor definition:
+
[source,subs="+attributes"]
----
CHE_EDITOR_CONCEAL_CONFIGMAP_NAME=che-conceal-$CHE_EDITOR_CONCEAL_FILE_NAME
----

. Create the ConfigMap:
+
[source,subs="+attributes"]
----
{orch-cli} create configmap $CHE_EDITOR_CONCEAL_CONFIGMAP_NAME \
    --namespace $OPERATOR_NAMESPACE \
    --from-literal=$CHE_EDITOR_CONCEAL_FILE_NAME=""
----

> . Find out the Operator subscription namespace if it exists:

Reviewer: ```suggestion
. Find out the Operator subscription namespace (if it exists):
```

+
[source,subs="+attributes"]
----
SUBSCRIPTION_NAMESPACE=$({orch-cli} get subscription \
    --all-namespaces \
    --field-selector=metadata.name={prod-operator-subscription} \
    --output jsonpath='{.items[0].metadata.namespace}' 2>/dev/null
)
----

> . Patch the {kubernetes} resource to mount the ConfigMap with the empty editor definition. The resource to patch depends on the existence of the Operator subscription. If the subscription exists, then the subscription should be patched, otherwise the Operator deployment:

Reviewer: ```suggestion
. Patch the {kubernetes} resource to mount the ConfigMap with the empty editor definition. The resource to patch depends on the existence of the Operator subscription. If the subscription exists, then the subscription should be patched. If not, patch the Operator deployment:
```
